dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt

```diff
@@ -53,58 +64,141 @@ class DataManager(
+            logger.info(
+                "Received quality assurance for data upload with DataId: $dataId with Correlation Id: $correlationId",
+            )
+        } else {
+            val internalMessage = "Error updating metadata data. Correlation ID: $correlationId"
+            logger.error(internalMessage)
+            throw InternalServerErrorApiException(
+                "Update of meta data failed", "The update of the metadataset failed",
+                internalMessage,
+            )
+        }
+    }
+
+    /**
+     * This method retrieves data from the temporary storage
+     * @param dataId is the identifier for which all stored data entries in the temporary storage are filtered
+     * * @return stringified data entry from the temporary store
+     */
+    fun selectDataSetFromTemporaryStorage(dataId: String): String {
+        val rawValue = dataInformationHashMap.getOrElse(dataId) {
+            throw ResourceNotFoundApiException(
+                "Data ID not found in temporary storage",
+                "Dataland does not know the data id $dataId",
+            )
+        }
+        return objectMapper.writeValueAsString(rawValue)
+    }
+
+    /**
+     * Method to temporarily store a data set in a hash map and send a message to the storage_queue
+     * @param storableDataSet The data set to store
+     * @param companyName The name of the company corresponding to the data set to store
+     * @param correlationId The correlation id of the request initiating the storing of data
+     * @return ID of the stored data set
+     */
+    fun storeDataSetInTemporaryStoreAndSendDataReceivedMessage(
         storableDataSet: StorableDataSet,
         companyName: String,
         correlationId: String,
     ): String {
-        val dataId: String
-        try {
-            dataId = storageClient.insertData(correlationId, objectMapper.writeValueAsString(storableDataSet)).dataId
-        } catch (e: ServerException) {
-            val internalMessage = "Error storing data." +
-                " Received ServerException with Message: ${e.message}. Correlation ID: $correlationId"
+        val dataId = generateRandomDataId()
+        dataInformationHashMap[dataId] = objectMapper.writeValueAsString(storableDataSet)
+        cloudEventMessageHandler.buildCEMessageAndSendToQueue(
+            dataId, "Data to be stored", correlationId,
+            storageQueue,
+        )
+        logger.info(
+            "Stored StorableDataSet of type ${storableDataSet.dataType} for company ID in temporary store" +
+                "${storableDataSet.companyId}. Company Name $companyName received ID $dataId from storage. " +
+                "Correlation ID: $correlationId.",
+        )
+        return(dataId)
+    }
+
+    /**
+     * Method to generate a random Data ID
+     * @return generated UUID
+     */
+    fun generateRandomDataId(): String {
+        return "${UUID.randomUUID()}:${UUID.randomUUID()}_${UUID.randomUUID()}"
```

Comment:
shouldn't we simplify this to be simpliy UUID?

Reply:
I don't know. I assume there was reason for this. Maybe to "eliminate" the possibility of colliding ids.

Reply:
I don't have a strong opionion on this. Does ist hurt to have longer IDs?

Reply:
I think we had this pattern for stange eurodat-reasons.
We don't need it anymore.

Can you simply try a single UUID an check whether this works?
(There are no strong reasons against longer IDs, but there are also absolutely none in favor of it)

Reply:
UUID shortend

Reply:
ok